### PR TITLE
Added ability to add and delete a task

### PR DESCRIPTION
**Pull request by RyhorKrs**
Added ability to add and delete a task, cleaned code

### CODE REVIEW

File: src/modules/app/tasks/tasks.component.ts [hunk 1]
```diff
@@ -26,27 +26,14 @@ export class TasksComponent implements OnInit, OnDestroy {
   constructor (private fbService: FbAuthService) {}
 
   public ngOnInit() {
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:21]: void

File: src/modules/app/tasks/tasks.component.ts [hunk 2]
```diff
@@ -26,27 +26,14 @@ export class TasksComponent implements OnInit, OnDestroy {
...
-    this.fbService.uid = JSON.parse(localStorage.uid);
     this.getUserContent();
   }
 
-  public ngOnDestroy() {
-    this.fbService.uid = '';
-    this.fbService.id = '';
-  }
-
   public user: any = {}
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts]: ;

File: src/modules/app/header/header.component.ts [hunk 3]
```diff
@@ -13,8 +13,7 @@ export class HeaderComponent implements OnInit {
   public isSignIn: boolean = false;
   public showUserMenu: boolean = false;
   public language: string = 'en';
-
-  public currentUser: string | null = '123';
+  public currentUser: string = 'xxx';
 
   constructor(
```
Comment (AliakseiKliashchou) [inline src/modules/app/header/header.component.ts:18]: space

File: src/modules/app/tasks/tasks.component.ts [hunk 4]
```diff
@@ -9,41 +10,49 @@ import { Task } from './../../../shared/interfaces/TASK';
...
+    ) {}
 
   public ngOnInit() {
     this.getUserContent();
   }
 
-  public user: any = {}
-
   public getUserContent(): void {
     this.fbService.getDataFromDb(JSON.parse(localStorage.uid)).subscribe(user => {
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:26]: add model for argument 'user'

File: src/shared/guards/signinout.guard.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { LocalStorageService } from '../services/localStorage.service';
+
+@Injectable({providedIn: 'root'})
+export class SignInOutGuard implements CanActivate {
+
+  constructor(
```
Comment (AliakseiKliashchou) [inline src/shared/guards/signinout.guard.ts:9]: space

File: src/shared/guards/signinout.guard.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    private localStorageService: LocalStorageService,
+    private router: Router
+  ) {}
+
+  canActivate(
```
Comment (AliakseiKliashchou) [inline src/shared/guards/signinout.guard.ts:14]: space

File: src/shared/guards/signinout.guard.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+    ): Observable<boolean> | Promise<boolean> | boolean {
+      if(!this.localStorageService.getItem('uid')) {
+
```
Comment (AliakseiKliashchou) [inline src/shared/guards/signinout.guard.ts:19]: useless empty line

File: src/shared/interfaces/TASK.ts [hunk 8]
```diff
@@ -1,5 +1,6 @@
 export interface Task {
   taskTitle: string,
-  taskDescription: string
+  taskDescription: string,
+  id?: string
```
Comment (AliakseiKliashchou) [inline src/shared/interfaces/TASK.ts:4]: ;

File: src/shared/services/fbAuth.service.ts [hunk 9]
```diff
@@ -39,7 +37,7 @@ export class FbAuthService {
 
   public getDataFromDb(uid: string): Observable<any> {
     return this.http
-      .get(`https://angular-todo-7e025-default-rtdb.firebaseio.com/users/${uid}.json`)
+      .get(`${fbUrl}/users/${uid}.json`)
       .pipe(map(res => {
         return res;
```
Comment (AliakseiKliashchou) [inline src/shared/services/fbAuth.service.ts:42]: one code line

File: src/modules/app/header/header.component.ts (hunk 3 above)
Comment (RyhorKrs) [reply src/modules/app/header/header.component.ts:18]: what kind of space?

**Review by AliakseiKliashchou — APPROVED** [review]

**Pull request merged**